mongodbatlas/resource_mongodbatlas_cluster_outage_simulation.go

```diff
@@ -0,0 +1,193 @@
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"outage_filters": {
+				Type:     schema.TypeList,
+				MinItems: 1,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cloud_provider": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"region_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"start_request_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"simulation_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceMongoDBClusterOutageSimulationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID := d.Get("project_id").(string)
+	clusterName := d.Get("cluster_name").(string)
+
+	requestBody := &matlas.ClusterOutageSimulationRequest{
+		OutageFilters: newOutageFilters(d),
+	}
+
+	_, _, err := conn.ClusterOutageSimulation.StartOutageSimulation(ctx, projectID, clusterName, requestBody)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(errorClusterOutageSimulationCreate, projectID, clusterName, err))
+	}
+	d.SetId(encodeStateID(map[string]string{
+		"project_id":   projectID,
+		"cluster_name": clusterName,
+	}))
+
+	return resourceMongoDBAClusterOutageSimulationRead(ctx, d, meta)
+}
+
+func newOutageFilters(d *schema.ResourceData) []matlas.ClusterOutageSimulationOutageFilter {
+	outageFilters := make([]matlas.ClusterOutageSimulationOutageFilter, len(d.Get("outage_filters").([]interface{})))
+
+	for k, v := range d.Get("outage_filters").([]interface{}) {
+		a := v.(map[string]interface{})
+		outageFilters[k] = matlas.ClusterOutageSimulationOutageFilter{
+			CloudProvider: pointy.String(a["cloud_provider"].(string)),
+			RegionName:    pointy.String(a["region_name"].(string)),
+			Type:          pointy.String(a["type"].(string)),
+		}
+	}
+
+	return outageFilters
+}
+
+func resourceMongoDBAClusterOutageSimulationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+	ids := decodeStateID(d.Id())
+	projectID := ids["project_id"]
+	clusterName := ids["cluster_name"]
+
+	outageSimulation, resp, err := conn.ClusterOutageSimulation.GetOutageSimulation(ctx, projectID, clusterName)
+
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf(errorClusterOutageSimulationRead, projectID, clusterName, err))
+	}
+
+	err = convertOutageSimulationToSchema(outageSimulation, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
I am not a fan of changing the values of a pointer `d` inside a function as I think it makes the code less readable (it is not clear you are changing `d`). The recommended way is to return a new object and set it in the calling function similar to what you are doing with `newOutageFilters`. (Not a blocker, just something to think about)